src/EppoClient.php

```diff
@@ -361,16 +377,183 @@ private function getAssignmentDetail(
+     * @param array<string>|array<string, array<string, ?object>>|array<string, AttributeSet> $actions
+     * @param string $defaultVariation
+     * @return BanditResult
+     *
+     * @throws EppoClientException
+     */
+    public function getBanditAction(
+        string $flagKey,
+        string $subjectKey,
+        array|AttributeSet $subjectContext,
+        array $actions,
+        string $defaultVariation
+    ): BanditResult {
+        try {
+            // Normalize the subject and action into AttributeSets. These functions detect the structure of the
+            // user data allowing for either automatic or manual sorting into numeric and categorical attributes.
+            $subject = AttributeSet::fromFlexibleInput($subjectContext);
+            $actionContexts = AttributeSet::arrayFromFlexibleInput($actions);
+
+            return $this->getBanditDetail($flagKey, $subjectKey, $subject, $actionContexts, $defaultVariation);
+        } catch (EppoException $e) {
+            if ($this->isGracefulMode) {
+                error_log('[Eppo SDK] Error selecting bandit action: ' . $e->getMessage());
+                return new BanditResult($defaultVariation);
+            } else {
+                throw EppoClientException::from($e);
+            }
+        }
+    }
+
+    /**
+     * @param string $flagKey
+     * @param string $subjectKey ,
+     * @param AttributeSet $subject
+     * @param array<string, AttributeSet> $actionsWithContext
+     * @param string $defaultVariation
+     * @return BanditResult
+     *
+     * @throws InvalidConfigurationException
+     * @throws HttpRequestException
+     * @throws InvalidApiKeyException
+     * @throws InvalidArgumentException
+     * @throws EppoClientException
+     * @throws BanditEvaluationException
+     */
+    private function getBanditDetail(
+        string $flagKey,
+        string $subjectKey,
+        AttributeSet $subject,
+        array $actionsWithContext,
+        string $defaultVariation
+    ): BanditResult {
+        Validator::validateNotBlank($flagKey, 'Invalid argument: flagKey cannot be blank');
+
+        $isBanditFlag = $this->configurationLoader->isBanditFlag($flagKey);
+
+        if (empty($actionsWithContext) && $isBanditFlag) {
+            // This exception is caught in graceful mode and the default is returned (@see getBanditAction)
+            throw new BanditEvaluationException("No actions provided for bandit flag {$flagKey}");
+        }
+
+        $variation = $this->getStringAssignment(
+            $flagKey,
+            $subjectKey,
+            $subject->toArray(),
+            $defaultVariation
+        );
+
+        if (!$isBanditFlag) {
+            // It's likely that the developer made a mistake passing a non-bandit flag so let's warn them.
+            syslog(LOG_WARNING, "[Eppo SDK]: Flag \"{$flagKey}\" does not contain a Bandit");
+
+            // Return the computed variation without doing any more Bandit work.
```

Comment:
This may not be a mistake. A bandit test could be reverted/rolled out in the UI. I think we ditch the warning.

Reply:
aaand it's gone